pkg/kubernetes/queue.go

```diff
@@ -0,0 +1,145 @@
+package kubernetes
+
+import (
+	"context"
+	"log"
+	"strings"
+	"time"
+
+	pr "github.com/kyverno/policy-reporter/pkg/crd/api/policyreport/v1alpha2"
+	"github.com/kyverno/policy-reporter/pkg/crd/client/clientset/versioned/typed/policyreport/v1alpha2"
+	"github.com/kyverno/policy-reporter/pkg/report"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+type Queue struct {
+	queue     workqueue.RateLimitingInterface
+	client    v1alpha2.Wgpolicyk8sV1alpha2Interface
+	debouncer Debouncer
+}
+
+func (q *Queue) Add(event report.Event, obj *v1.PartialObjectMetadata) error {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		return err
+	}
+
+	q.queue.Add(event.String() + ":" + key)
```

Comment:
Not sure about that, aren't you going to loose advantages of the queue here ?

Reply:
Which advantage do you mean? After each informer event it adds the key and related event type to the queue und process it in the worker.

Reply:
The queue guarantees that for a given item in the queue, only one worker will process it at a time.
This means that if you receive multiple events, they get pushed to the queue but if a worker is already processing the item, it is requeued and processed again later.
For this to work the key has to be stable (all events triggered by the same item need to produce the same key).

Reply:
Ah okay, understand. So I need some kind of cache to get information about the type of underlying informer event

Reply:
hm yes, thats right. I will try it out. Thanks a lot as always.